### PR TITLE
[JENKINS-27426] record premerge branch and commit hash

### PR DESCRIPTION
**Pull request by mattfirtion**

### CODE REVIEW

File: src/main/java/hudson/plugins/git/GitSCM.java [hunk 1]
```diff
@@ -1016,7 +1021,14 @@ public void checkout(Run<?, ?> build, Launcher launcher, FilePath workspace, Tas
         environment.put(GIT_COMMIT, revToBuild.revision.getSha1String());
         Branch branch = Iterables.getFirst(revToBuild.revision.getBranches(),null);
         if (branch!=null) { // null for a detached HEAD
-            environment.put(GIT_BRANCH, getBranchName(branch));
+        	environment.put(GIT_BRANCH, getBranchName(branch));
```
Comment (KostyaSha) [inline src/main/java/hudson/plugins/git/GitSCM.java]: indentation change

File: src/main/java/hudson/plugins/git/GitSCM.java [hunk 2]
```diff
@@ -165,7 +164,13 @@ public GitSCM(
...
 
         // moved from createBranches
-        this.branches = isEmpty(branches) ? newArrayList(new BranchSpec("*/master")) : branches;
+        if (branches == null) {
+            branches = new ArrayList<BranchSpec>();
+        }
+        if (branches.isEmpty()) {
+            branches.add(new BranchSpec("*/master"));
+        }
+        this.branches = branches;
```
Comment (KostyaSha) [inline src/main/java/hudson/plugins/git/GitSCM.java]: why it was changed?

**Comment by KostyaSha** [comment]
Why not contribute this vars from pre-merge extension?

**Comment by KostyaSha** [comment]
imho if variable is expected to be universal to all plugins, that it makes to call it intact `GIT_MARKED_REVISION` if it only from/for premerge, then it should be contributed from pre-merge extension with `GIT_PREMERGE_*`

**Comment by mattfirtion** [comment]
I agree with the naming.  EnvVars needs an AbstractBuild.  There is not an
AbstractBuild available in PreBuildMerge ext.

I'm new here, so any advice is welcome.

On Sat, Mar 14, 2015 at 11:47 AM, Kanstantsin Shautsou <
notifications@github.com> wrote:

> imho if variable is expected to be universal to all plugins, that it makes
> to call it intact GIT_MARKED_REVISION if it only from/for premerge, then
> it should be contributed from pre-merge extension with GIT_PREMERGE_*
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/git-plugin/pull/311#issuecomment-80543578>.
>



-- 
Matt Firtion

**Comment by mattfirtion** [comment]
Closing pull request based on feedback.

**Pull request closed without merging**